test/functional/eval/wait_spec.lua

```diff
@@ -0,0 +1,40 @@
+local helpers = require('test.functional.helpers')(after_each)
+local clear, eval, eq, feed, feed_command, next_msg, nvim, source =
+helpers.clear, helpers.eval, helpers.eq, helpers.feed, helpers.feed_command,
+helpers.next_msg, helpers.nvim, helpers.source
+
+before_each(function()
+  clear()
+  local channel = nvim('get_api_info')[1]
+  nvim('set_var', 'channel', channel)
+end)
+
+describe('wait()', function()
+  it('waits and returns 0 when condition is satisfied', function()
+    source([[
+    let g:_awake = 0
+    call timer_start(1000, { -> nvim_command('let g:_awake = 1') })
```

Comment:
10 or 100 would be better, tests should be as fast as possible. The `wait()` timeout below can stay at 1500 though, for slow environments like ASAN.